test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs

```diff
@@ -4768,6 +4768,693 @@ public Task TestPacManGetUnrestoredPackagesByDependencyOrder()
+            using (var testSolutionManager = new TestSolutionManager(true))
+            {
+                var testSettings = new Configuration.NullSettings();
+                var token = CancellationToken.None;
+                var deleteOnRestartManager = new TestDeleteOnRestartManager();
+                var nuGetPackageManager = new NuGetPackageManager(
+                    sourceRepositoryProvider,
+                    testSettings,
+                    testSolutionManager,
+                    deleteOnRestartManager);
+
+                var installationCompatibility = new Mock<IInstallationCompatibility>();
+                nuGetPackageManager.InstallationCompatibility = installationCompatibility.Object;
+
+                var msBuildNuGetProject = testSolutionManager.AddNewMSBuildProject();
+                var msBuildNuGetProjectSystem = msBuildNuGetProject.MSBuildNuGetProjectSystem as TestMSBuildNuGetProjectSystem;
+                var packagesConfigPath = msBuildNuGetProject.PackagesConfigNuGetProject.FullPath;
+                var packageIdentity = PackageWithDependents[0];
+
+                // batch handlers
+                var batchStartId = string.Empty;
+                var batchEndId = string.Empty;
+                var projectName = string.Empty;
+
+                // add batch events handler
+                nuGetPackageManager.BatchStart += (o, args) =>
+                {
+                    batchStartId = args.Id;
+                    projectName = args.Name;
+                };
+
+                nuGetPackageManager.BatchEnd += (o, args) =>
+                {
+                    batchEndId = args.Id;
+                };
+
+                // Act
+                await nuGetPackageManager.InstallPackageAsync(msBuildNuGetProject, packageIdentity,
+                    new ResolutionContext(), new TestNuGetProjectContext(), sourceRepositoryProvider.GetRepositories().First(), null, token);
+
+                // Assert
+                // Check that the packages.config file exists after the installation
+                Assert.True(File.Exists(packagesConfigPath));
+
+                // Check the number of packages and packages returned by PackagesConfigProject after the installation
+                var packagesInPackagesConfig = (await msBuildNuGetProject.PackagesConfigNuGetProject.GetInstalledPackagesAsync(token)).ToList();
+                Assert.Equal(1, packagesInPackagesConfig.Count);
+                Assert.Equal(packageIdentity, packagesInPackagesConfig[0].PackageIdentity);
+                Assert.NotEmpty(batchStartId);
+                Assert.NotEmpty(batchEndId);
+                Assert.Equal(batchStartId, batchEndId);
+
+                // check project name
+                string expectedProjectName;
+                msBuildNuGetProject.TryGetMetadata(NuGetProjectMetadataKeys.Name, out expectedProjectName);
```

Comment:
Set the project name in AddNewMSBuildProject and verify it is exactly as expected here instead of reading it from the project, it will be much safer validation.